src/plugins/data_source_management/server/routes/data_connections_router.ts

```diff
@@ -222,7 +222,9 @@ export function registerDataConnectionsRoute(router: IRouter, dataSourceEnabled:
         let dataConnectionsresponse;
         if (dataSourceEnabled && dataSourceMDSId) {
           const client = await context.dataSource.opensearch.legacy.getClient(dataSourceMDSId);
-          dataConnectionsresponse = await client.callAPI('ppl.getDataConnections');
+          dataConnectionsresponse = await client.callAPI('ppl.getDataConnections', {
+            requestTimeout: 2000, // Enforce timeout to avoid hanging requests
```

Comment:
this is a normal opensearch call, do you know why it's timing out? is it when cluster is deleted or not reachable but the MDS reference still exists? also 2 seconds might be too small